functions/Get-DbaMsdtc.ps1

```diff
@@ -90,77 +94,73 @@ Get DTC status for the computer srv0042 and show in a grid view
-					Select-Object PSComputerName, DisplayName, State, status, StartMode, StartName
-				}
-				catch
-				{
-					Write-Warning "Can't connect to CIM on $Computer"
-				}
-			}
-			$reg = $CIDs = $null
-			$CIDHash = @{ }
-			try
-			{
-				Write-Verbose "Getting Registry Values on $Computer"
-				$reg = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCSecurity -ErrorAction Stop
-				$CIDs = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCCIDs -ErrorAction Stop
-				foreach ($key in $CIDs) { $CIDHash.Add($key.Data, $key.CID) }
-			}
-			catch
-			{
-				Write-Warning "Can't connect to registry on $Computer"
-			}
-			if ($dtcservice)
-			{
-				[PSCustomObject]@{
-					ComputerName = $dtcservice.PSComputerName
-					DTCServiceName = $dtcservice.DisplayName
-					DTCServiceState = $dtcservice.State
-					DTCServiceStatus = $dtcservice.Status
-					DTCServiceStartMode = $dtcservice.StartMode
-					DTCServiceAccount = $dtcservice.StartName
-					DTCCID_MSDTC = $CIDHash['MSDTC']
-					DTCCID_MSDTCUIS = $CIDHash['MSDTCUIS']
-					DTCCID_MSDTCTIPGW = $CIDHash['MSDTCTIPGW']
-					DTCCID_MSDTCXATM = $CIDHash['MSDTCXATM']
-					networkDTCAccess = $reg.networkDTCAccess
-					networkDTCAccessAdmin = $reg.networkDTCAccessAdmin
-					networkDTCAccessClients = $reg.networkDTCAccessClients
-					networkDTCAccessInbound = $reg.networkDTCAccessInbound
-					networkDTCAccessOutBound = $reg.networkDTCAccessOutBound
-					networkDTCAccessTip = $reg.networkDTCAccessTip
-					networkDTCAccessTransactions = $reg.networkDTCAccessTransactions
-					XATransactions = $reg.XATransactions
-				}
-			}
-		}
+    ForEach ($Computer in $Computername)
+    {
+      $dtcservice = $null
+      Write-Verbose "$functionName - Getting DTC on $computer via WinRM"
+      $dtcservice = Get-Ciminstance -ComputerName $Computer -Query $query
+      if ( $null -eq $dtcservice )
+      {
+          Write-Verbose "$functionName - Failed To get DTC via WinRM. Getting DTC on $computer via DCom"
+          $SessionParams = @{ }
+          $SessionParams.ComputerName = $Computer
+          $SessionParams.SessionOption = (New-CimSessionOption -Protocol Dcom)
+          $Session = New-CimSession @SessionParams
+          $dtcservice = Get-Ciminstance -CimSession $Session -Query $query
+      }
+      if ( $null -eq $dtcservice )
+      {
+          Write-Warning "$functionName - Can't connect to CIM on $Computer"
+      }
+      $reg = $CIDs = $null
+      $CIDHash = @{ }
+      Write-Verbose "$functionName - Getting MSDTC Security Registry Values on $Computer"
+      $reg = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCSecurity
+      if ( $null -eq $reg )
+      {
+        Write-Warning "$functionName - Can't connect to MSDTC Security registry on $Computer"
+      }
+      Write-Verbose "$functionName - Getting MSDTC CID Registry Values on $Computer"
+      $CIDs = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCCIDs

```

Comment:
Actually, I was thinking about catching whenever the previous command was _unable to connect at all_. In that case it is safe to assume the second remoting attempt will fail too.
In that scenario you'd need to make sure the first script will not be throwing errors, in order to avoiding muddling the water (a $null return should work).
Then you call the first Invoke-Command with try/catch and on catch disable the second attempt.